client/landing/gutenboarding/stores/onboard/reducer.ts

```diff
@@ -43,7 +43,21 @@ const siteVertical: Reducer<
 	return state;
 };
 
-const reducer = combineReducers( { domain, siteTitle, siteVertical } );
+const pageLayouts: Reducer<
+	Array< string >,
+	ReturnType< typeof Actions[ 'togglePageLayout' ] >
+> = ( state = [], action ) => {
+	if ( action.type === ActionType.TOGGLE_PAGE_LAYOUT ) {
+		const layout = action.pageLayout;
+		if ( state.includes( layout.slug ) ) {
+			return state.filter( item => item !== layout.slug );
+		}
```

Comment:
p.s. (probably worth ignoring) but would it help to reduce this down to `O(N)`, or a single pass of the array? Something like:

```
const filtered = state.filter( item => item !== layout.slug );

if ( state.length === filtered.length ) {
    return [ ...state, item ];
}

return filtered;
```

or is that just harder to read with little extra value? Maybe there's a cleaner approach?

Reply:
This probably isn't worth optimizing much.

Reply:
Heh I guess suggesting to improve the state tree into an indexed form may sound even less indulging.